File: app/src/main/java/co/medellinandroid/spotifyartists/rest/ApiInterface.java
package co.medellinandroid.spotifyartists.rest;

import co.medellinandroid.spotifyartists.model.SpotifyArtistsResponse;
import co.medellinandroid.spotifyartists.model.SpotifySongsResponse;
import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiInterface {

  @GET("search") Observable<SpotifyArtistsResponse> getTopTenArtists(
      @Query(value = "q", encoded = true) String query, @Query("type") String type,
      @Query("market") String market, @Query("limit") int limit);

  @GET("artists/{id}/top-tracks") Observable<SpotifySongsResponse> getTopTenSongsByArtist(
      @Path("id") String artistId, @Query("country") String country);
}

File: app/src/main/java/co/medellinandroid/spotifyartists/rest/ApiInterface.kt
package co.medellinandroid.spotifyartists.rest

import co.medellinandroid.spotifyartists.model.SpotifyArtistsResponse
import co.medellinandroid.spotifyartists.model.SpotifySongsResponse
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface ApiInterface {

  @GET("search")
  fun getTopTenArtists(
      @Query(value = "q", encoded = true) query: String, @Query("type") type: String,
      @Query("market") market: String, @Query(
          "limit") limit: Int): Observable<SpotifyArtistsResponse>

  @GET("artists/{id}/top-tracks")
  fun getTopTenSongsByArtist(
      @Path("id") artistId: String, @Query(
          "country") country: String): Observable<SpotifySongsResponse>
}
